_posts/2020-03-02-reduce-boilerplate-with-records.md

```diff
@@ -0,0 +1,212 @@
+---
+layout: post
+title: Reduce boilerplate in Java backends with records
+author: janneri
+excerpt: >
+  Java 14 (March 2020) introduces records as a preview feature. Records provide a compact syntax for declaring simple, immutable data carrier classes, such as Point(x: int, y: int). This blog gives an example of records in rest/json APIs.
+tags:
+ - Java
+ - Java 14
+ - Records
+ - JEP 359
+---
+
+# Reduce boilerplate in Java backends with records
+
+# Key takeaways
+- Java 14 (March 2020) introduces [records](https://openjdk.java.net/jeps/359) as a preview feature
+- Records provide a compact syntax for declaring simple, immutable data carrier classes, such as ```Point(x: int, y: int) {}```
+- This blog is about records and rest/json APIs
+- Frameworks can easily translate values from and to json using records
+- If you want builder-pattern, you still need (IDE or annotation-based) code generation    
+
+# Introduction to Java records
+
+![Java Versions](/img/java14-records/java-versions.png)
+
+As illustrated in the picture above, Java 14 released in March 2020 introduces a new interesting feature named [records](https://openjdk.java.net/jeps/359). 
+This is how records are declared:
+
+```
+record Point(x: int, y: int) {}
+```
+
+A record automatically acquires:
+- a public constructor
+- accessors to the fields
+- equals() and hashCode()
+- toString()
+
+Previously IDEs and annotation-based code generation have helped us generate these members.
```

Comment:
Previously, IDEs and annotation-based code generation have helped us generate these class members.